File: backend/src/main/java/com/btalk/entity/Message.java
package com.btalk.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import com.btalk.constants.MessageType;

import java.time.LocalDateTime;

@Entity
@Table(name = "messages")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String messageId;

    @Column(name = "conversation_id", nullable = false)
    private String conversationId;

    @Column(name = "sender_id", nullable = false)
    private String senderId;

    private String content;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MessageType messageType;

    @CreationTimestamp
    private LocalDateTime sentAt;

    @Column(name = "call_duration")
    private Integer callDuration; // Duration in seconds for call messages

    @Column(name = "call_type")
    @Enumerated(EnumType.STRING)
    private com.btalk.constants.CallType callType; // Type of call for call messages

    @Column(name = "call_status")
    private String callStatus; // Status of the call (MISSED, ENDED, REJECTED)
}

File: frontend/src/app/models/message.ts
import { Attachment } from "./attachment";

export enum MessageType {
  TEXT = 'TEXT',
  IMAGE = 'IMAGE',
  VIDEO = 'VIDEO',
  DOCUMENT = 'DOCUMENT',
  AUDIO = 'AUDIO',
  CALL = 'CALL'
}

export interface Message {
  messageId?: string | null;  // UUID as string, optional because it's assigned by server
  conversationId: string; // UUID as string
  senderId: string; // UUID as string
  senderName?: string | null;  // Optional, might be added by server
  content: string;
  messageType: MessageType;
  sentAt: Date;     // Will be set by server
  status: 'SENT' | 'DELIVERED' | 'SEEN';
  attachments: Attachment[];
  callDuration?: number; // Duration in seconds for call messages
  callType?: 'AUDIO' | 'VIDEO'; // Type of call for call messages
  callStatus?: 'MISSED' | 'ENDED' | 'REJECTED'; // Status of the call
}
